test/integration/options/watch.spec.js

```diff
@@ -111,50 +63,98 @@ describe('--watch', function() {
       touchFile(gitFile);
       touchFile(nodeModulesFile);
 
-      const [mocha, outputPromise] = runMochaJSONWatchAsync(
-        [this.testFile, '--extension', 'xyz,js'],
-        {
-          cwd: this.tempDir
+      return runMochaWatch(
+        [testFile, '--extension', 'xyz,js'],
+        this.tempDir,
+        () => {
+          touchFile(gitFile);
+          touchFile(nodeModulesFile);
         }
-      );
+      ).then(results => {
+        expect(results, 'to have length', 1);
+      });
+    });
 
-      return expect(
-        sleep(1000)
-          .then(() => {
-            touchFile(gitFile);
-            touchFile(nodeModulesFile);
-          })
-          .then(() => sleep(1000))
-          .then(() => {
-            mocha.kill('SIGINT');
-            return outputPromise;
-          }),
-        'when fulfilled',
-        'to have length',
-        1
-      );
+    it('reloads test files when they change', function() {
+      const testFile = path.join(this.tempDir, 'test.js');
+      copyFixture('options/watch/test-file-change', testFile);
+
+      return runMochaWatch(
+        [testFile, '--watch-files', '**/*.js'],
```

Comment:
We don't have a `--watch-files` flag, yet. Why does this test pass then?

Reply:
Oops. This is an artifact of extracting it from my wip branch. It does not fail the tests because we use the `--extensions` option to determine which files to watch and the default value is `js`. We also don’t use yarg’s [`strict`option](https://github.com/yargs/yargs/blob/master/docs/api.md#strictenabledtrue), something that I would encourage. This makes yargs silently accept unknown flags.

I’ve removed the artifact.